tower/src/buffer/layer.rs

```diff
@@ -9,12 +9,13 @@ use tower_service::Service;
 /// which means that this layer can only be used on the Tokio runtime.
 ///
 /// See the module documentation for more details.
-pub struct BufferLayer<Request> {
+pub struct BufferLayer<Request, E2> {
     bound: usize,
     _p: PhantomData<fn(Request)>,
+    _e: PhantomData<E2>,
```

Comment:
This should be in a `fn` so that it doesn't effect the service's `Send`/`Sync`/etc impls.